gtsfm/loader/loader_base.py

```diff
@@ -122,10 +142,27 @@ def get_valid_pairs(self) -> List[Tuple[int, int]]:
             list of valid index pairs.
         """
         indices = []
-
         for idx1 in range(self.__len__()):
             for idx2 in range(self.__len__()):
                 if self.validate_pair(idx1, idx2):
                     indices.append((idx1, idx2))
 
         return indices
+
+    def get_valid_triplets(self) -> List[Tuple[int, int, int]]:
+        """Get the valid triplets of images.
+
+        Returns:
+            list of indices of valid triplets.
+        """
+        indices = []
+        for idx1, idx2 in itertools.product(range(len(self)), range(len(self))):
```

Comment:
are permutations considered unique?
```python
>>> list(itertools.product( range(3), range(3) ))
[(0, 0), (0, 1), (0, 2), (1, 0), (1, 1), (1, 2), (2, 0), (2, 1), (2, 2)]
```
like (0,1) and (1,0)?

Reply:
Yes they would be considered unique, because I am relying on the `validate_pair` function. That should return true only for one of (0, 1)/(1, 0).

Reply:
got it -- do you mind adding that to the docstring?

Reply:
(or as an inline comment)